### PR TITLE
Add doc on custom IPXE features:

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

This helps to increase the understanding of the features we enable in our IPXE builds.
Just a start.
More breakdowns by arch to follow.

## Why is this needed

Fixes: #

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

File: docs/IPXE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+See the `.h` files in the [ipxe/ipxe/](../ipxe/ipxe/) directory for the exact details of the features enabled.
+
+- image trust commands like [imgtrust](https://ipxe.org/cmd/imgtrust), [imgverify](https://ipxe.org/cmd/imgverify)
+- boot from a web server via [HTTP/HTTPS](https://ipxe.org/cmd)
+- [san](https://ipxe.org/cmd) commands support
+- IPv6 support
+- [ntp](https://ipxe.org/cmd/ntp) command support
+- VLAN([vcreate](https://ipxe.org/cmd/vcreate),[vdestroy](https://ipxe.org/cmd/vdestroy)) command support
+- [params](https://ipxe.org/cmd/params) command support
+- [param](https://ipxe.org/cmd/param) command support
```
Comment (mmlb) [inline docs/IPXE.md:18]: These go hand in hand, I'd put them in just one line.

**Review by mmlb — CHANGES_REQUESTED** [review]

File: docs/IPXE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# IPXE
+
+In Boots we build custom IPXE binaries for different architectures.
```
Comment (mmlb) [inline docs/IPXE.md:3]: worth a call out to `make gen` and ../ipxev.mk?

Comment (jacobweinstock) [reply docs/IPXE.md:3]: `make gen` doesn't actually do any ipxe related things anymore now that we've moved to builtin `go:embed`.
Would you mind if I came back around to add details about `ipxev.mk`?

Comment (mmlb) [reply docs/IPXE.md:3]: oh darn I missed that then. How do we build ipxe binaries now?

Comment (jacobweinstock) [reply docs/IPXE.md:3]: `make ipxe` and `make build_all_ipxe` handle all that now.

Comment (jacobweinstock) [reply docs/IPXE.md:3]: the Go code embedding now happens at `go build` time.

File: docs/IPXE.md (hunk 1 above)
Comment (jacobweinstock) [reply docs/IPXE.md:18]: they were called out separately on https://ipxe.org/cmd so I added them here that way.

File: docs/IPXE.md (hunk 2 above)
Comment (mmlb) [reply docs/IPXE.md:3]: > Would you mind if I came back around to add details about `ipxev.mk`?

:+1:

Comment (mmlb) [reply docs/IPXE.md:3]: > `make ipxe` and `make build_all_ipxe` handle all that now.

Why get rid of iPXE building from `make gen`, I thought of it like a generic "generate all the deps" target.

**Review by mmlb — APPROVED** [review]

File: docs/IPXE.md (hunk 2 above)
Comment (jacobweinstock) [reply docs/IPXE.md:3]: that's fair. I didn't think of ipxe builds as being generated so I put it elsewhere. I like them on their own too for maintainability and composability.

**Pull request merged**